.azure/pipelines/helix-matrix.yml

```diff
@@ -1,6 +1,13 @@
 # We only want to run full helix matrix on main
-pr: none
 trigger: none
+pr:
+  branches:
+    include:
+    - main
+  paths:
+    include:
+    - src/Servers/IIS/**/*
+    - src/Servers/HttpSys/**/*
```

Comment:
src/Shared/HttpSys/
Or maybe the whole src/Shared/ folder since multiple things are used from there
https://github.com/dotnet/aspnetcore/blob/15f057051539cd2fd8d0b81542be346de83619c2/src/Servers/HttpSys/src/Microsoft.AspNetCore.Server.HttpSys.csproj#L26-L34